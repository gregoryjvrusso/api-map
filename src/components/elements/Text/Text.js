import React from 'react'
import styled, { css } from 'styled-components'
import { ifProp, theme } from 'styled-tools'
import { fontSize, fontWeight, margin } from 'styled-system'

const warning = css`
  color: ${theme('colors.texts.warning')};
`
const Text = styled.p`
  color: ${theme('colors.texts.primary')};
  ${fontSize}
  ${fontWeight}
  ${margin}
  ${ifProp('warning', warning)}
`

const TextDefault = ({ tag = 'p', text, ...others}) => {
  return (
    <Text 
      {...others} 
      as={tag}
    >
      {text}
    </Text>
  )
}

export default TextDefault