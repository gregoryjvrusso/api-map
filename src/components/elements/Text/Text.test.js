import React from 'react'
import { render } from '@testing-library/react'
import Text from './Text.js'

describe('Text component', () => {
  it('deve conseguir renderizar [snapshot]', async () => {
    const { container } = render(<Text text='Texto de teste' />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('deve conseguir renderizar [snapshot] como tag H1', async () => {
    const { container } = render(<Text text='Texto de teste 2' tag='h1' />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('deve conseguir renderizar [snapshot] como warning', async () => {
    const { container } = render(<Text text='Texto de teste' warning />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('deve conseguir renderizar [snapshot] como tag H1 e fontSize 15px', async () => {
    const { container } = render(<Text text='Texto de teste' tag='h1' fontSize='15px' />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
