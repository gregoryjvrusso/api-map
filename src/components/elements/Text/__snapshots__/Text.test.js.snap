// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component deve conseguir renderizar [snapshot] 1`] = `
<p
  class="sc-bdVaJa hCKHIO"
>
  Texto de teste
</p>
`;

exports[`Text component deve conseguir renderizar [snapshot] como tag H1 1`] = `
<h1
  class="sc-bdVaJa hCKHIO"
>
  Texto de teste 2
</h1>
`;

exports[`Text component deve conseguir renderizar [snapshot] como tag H1 e fontSize 15px 1`] = `
<h1
  class="sc-bdVaJa dsgeoZ"
  font-size="15px"
>
  Texto de teste
</h1>
`;

exports[`Text component deve conseguir renderizar [snapshot] como warning 1`] = `
<p
  class="sc-bdVaJa irvhcf"
>
  Texto de teste
</p>
`;
